server/src/main/java/org/opensearch/wlm/Rule.java

```diff
@@ -0,0 +1,370 @@
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Rule that = (Rule) o;
+        return Objects.equals(_id, that._id)
+            && Objects.equals(label, that.label)
+            && Objects.equals(feature, that.feature)
+            && Objects.equals(attributeMap, that.attributeMap)
+            && Objects.equals(updatedAt, that.updatedAt);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(_id, label, feature, attributeMap, updatedAt);
+    }
+
+    /**
+     * empty builder method for the {@link Rule}
+     * @return Builder object
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * builder method for the {@link Rule}
+     * @return Builder object
+     */
+    public Builder builderFromRule() {
+        return new Builder()._id(_id).label(label).feature(feature.getName()).updatedAt(updatedAt).attributeMap(attributeMap);
+    }
+
+    /**
+     * Builder class for {@link Rule}
+     */
+    @ExperimentalApi
+    public static class Builder {
+        private String _id;
+        private Map<RuleAttribute, Set<String>> attributeMap;
+        private Feature feature;
+        private String label;
+        private String updatedAt;
+
+        private Builder() {}
+
+        public static Builder fromXContent(XContentParser parser) throws IOException {
+            if (parser.currentToken() == null) {
+                parser.nextToken();
+            }
+
+            Builder builder = builder();
+            XContentParser.Token token = parser.currentToken();
+
+            if (token != XContentParser.Token.START_OBJECT) {
+                throw new IllegalArgumentException("Expected START_OBJECT token but found [" + parser.currentName() + "]");
+            }
+            Map<RuleAttribute, Set<String>> attributeMap1 = new HashMap<>();
+            String fieldName = "";
+            while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                if (token == XContentParser.Token.FIELD_NAME) {
+                    fieldName = parser.currentName();
+                } else if (token.isValue()) {
+                    if (fieldName.equals("_id")) {
+                        builder._id(parser.text());
+                    } else if (Feature.isValidFeature(fieldName)) {
+                        builder.feature(fieldName);
+                        builder.label(parser.text());
+                    } else if (fieldName.equals("updated_at")) {
+                        builder.updatedAt(parser.text());
+                    } else {
+                        throw new IllegalArgumentException(fieldName + " is not a valid field in Rule");
+                    }
+                } else if (token == XContentParser.Token.START_ARRAY) {
+                    RuleAttribute ruleAttribute = RuleAttribute.fromName(fieldName);
+                    Set<String> indexPatternList = new HashSet<>();
```

Comment:
This var name is coupled to a singled attribute based values, we should avoid that as this can easily create confusion. If there are other Rule attribute names then this is a misnomer